agents/appmotion_motor/sat1_motors_agent.py

```diff
@@ -0,0 +1,717 @@
+        motor = params.get('motor', 3)
+        with self.lock.acquire_timeout(1, job=f"getEncoderInfo_motor{motor}") as acquired:
+            if not acquired:
+                self.log.warn(f'Could not getEncoderInfo because lock held by {self.lock.job}')
+                return False, 'Could not acquire lock'                                    
+            ePositions = self.motors.retrieveEncoderInfo(motor)
+
+        return True, ("Current encoder positions: {}".format(ePositions),ePositions)    
+    
+    def getPositions(self, session, params = None):
+        """
+        Get the position of the motor in counts, relative to the set zero point (or starting point/home).
+        
+        Parameters:
+            params: {'motor': int, 'inches': bool}
+            motor: 1,2,3. Determines which motor, 3 is for all motors. (default 1)
+            inches: bool. Whether to return positions in inches or not. (default True)
+        Returns:
+            positions (list) -- the positions of the specified motors
+        """
+        
+        motor = params.get('motor', 1)
+        inches = params.get('inches', True)
+        with self.lock.acquire_timeout(1, job=f"getPositions_motor{motor}") as acquired:
+            if not acquired:
+                self.log.warn(f'Could not getPositions because lock held by {self.lock.job}')
+                return False, "Could not acquire lock"                                
+            if inches:
+                positions = self.motors.getPositionInInches(motor)
+            elif not inches:
+                positions = self.motors.getPosition(motor)
+            else: 
+                return False, "Invalid choice for inches parameter, must be boolean"
+
+        return True, "Current motor positions: {}".format(positions)
+
+    def posWhileMoving(self, session, params = None):
+        """
+        Get the position of the motor while it is currently in motion. An estimate based on the 
+        calculated trajectory of the movement, relative to the zero point.
+        
+        Parameters:
+            params: {'motor': int, 'inches': bool}
+            motor: 1,2,3. Determines which motor, 3 is for all motors. (default 1)
+            inches: bool. Whether to return positions in inches or not. (default True)
+        Returns:
+            positions (list) -- the positions of the specified motors
+        """
+        
+        inches = params.get('inches',True)
+        motor = params.get('motor', 1)
+        with self.lock.acquire_timeout(1, job=f"posWhileMoving_motor{motor}") as acquired:
+            if not acquired:
+                self.log.warn(f'Could not posWhileMoving because lock held by {self.lock.job}')
+                return False, "Could not acquire lock"                                
+            iPositions = self.motors.getImmediatePosition(motor,inches)
+
+        return True, "Current motor positions: {}".format(iPositions)
+
+    def isMoving(self, session, params = None):
```

Comment:
Added the move status to be published to feed

Reply:
If that's useful to record to disk, great. However I was more asking if this needs to be a task at all. Other tasks that need to check if the motors are currently moving do so directly. In practice, if you use this task to check for motion then it sounds like we should keep it around. If that is the case, the result of whether the motor is moving or not should be placed into the [session.data object](https://ocs.readthedocs.io/en/develop/developer/session-data.html).

Reply:
Yes, after our discussions I think the isMoving task should be placed in the session.data object. We do indeed use the `isMoving` task in our clients. 

I think it'd be worthwhile to replace all 'direct movement checks' with a call of this task instead, if only for consistency's sake. Though this will require just a little more thought than a single line replacement.

Reply:
Remy and I discussed this and agreed on moving the `self.move_status` check outside of the lock function to avoid lock conflicts, and keeping `is_moving` as a task. Also, we are now  returning the correct value for `self.move_status` by parsing the tuple that is returned in `is_moving`. i.e. `self.move_status = self.is_moving(motor)[1][1]`.